torchaudio/functional.py

```diff
@@ -406,6 +406,7 @@ def create_fb_matrix(n_freqs, f_min, f_max, n_mels, sample_rate):
         f_max (float): Maximum frequency (Hz)
         n_mels (int): Number of mel filterbanks
         sample_rate (int): Sample rate of the audio waveform
+        normalized (bool, optional): area-normalize the FB, so that each filter has equal area
```

Comment:
nit: "If True, normalize each filter to have equal area. If False, normalize each filter to have equal height.  (Default: ``False``)"

Thoughts?

Reply:
I love it! But before commiting these changes, I just learned about the [upcoming chainge in librosa.filters.mel](https://github.com/librosa/librosa/issues/1030) where the area normalizatioin has two variants, so people may choose from:

* `norm=None`: same as torchaudio's default MelScale
* `norm="slaney"`: the very new default (since 0.7.2, in Jan 13, 2020); same as the area normalization in this PR, dividing the filter by the width between the adjacent mel frequencies. [While this makes the L1 norm of each filter slightly different](https://github.com/librosa/librosa/issues/1030), this was preferred historically because it is regression tested to be bit identical with the MATLAB counterpart.
* `norm=1`: area normalization that has been the default historically and behaves exactly the same as `norm="slaney"`, but will change behavior in future librosa (likely 0.8), where the components will exactly sum to 1 (i.e. L1-normalized)
* `norm=p`: similar to the above, using Lp-norm, but there's no practical reason to use this.

Do you think we should follow their convention instead?

Reply:
That's good to know! Given how the implementation is just a small division at the end of the computation on the filterbanks, I was still considering something like `MelScale`, `SlaneyScale`. However, if we add a `MelPScale` with a parameter for p-norm, it seems like we are back to having to provide a parameter.

Let' just use their new convention.

Reply:
Sure, I'll modify the current PR to have librosa-0.8-style `norm=` options instead.

---

(Again, below not pertaining to this PR)

I still think the naming `SlaneyScale` on top of `MelScale` would be confusing as commented above, because:

* The Mel scale from the MATLAB auditory toolbox (referred to as "slaney") is also a Mel scale. It might give an impression that it is not, or less legitimate than the HTK Mel scale.
* What the current `MelScale` class represents is actually a Mel filter bank, because the term "Mel scale" simply refer to a series of Mel frequencies mapped from Hz (by a function that is roughly linear below 1kHz and roughly logarithmic above 1kHz). The `MelScale` class contains a set of triangular filters constructed from the Mel scale. So `MelFilterBank` would have been a better naming.
* With `MelScale` encapsulating both the scale AND the filterbank, it is not quite clear in the name `SlaneyScale` if "slaney" is referring to the scale or the normalization method, and we too seem to be confused throughout the comments on this PR.
* Nobody uses the term "slaney scale"; when googled with quotes, I see only three irrelevant results and none in Google Scholar.

Regardless of the future naming of `SlaneyScale` (or `mel_scale="htk"|"slaney"` as I'd prefer), it should only apply to the Mel scale, and it is orthogonal to the `norm` options this PR is addressing.